bugsnag-plugin-android-ndk/src/main/jni/jnicache.c

```diff
@@ -0,0 +1,249 @@
+//
+// Created by Karl Stenerud on 22.02.21.
+//
+
+#include "jnicache.h"
+#include <stdatomic.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <bugsnag_ndk.h>
+#include "safejni.h"
+
+static type_bsg_jni_cache jni_cache;
+static bool is_cache_valid = false;
+
+static inline jclass safe_find_class(JNIEnv *env, const char *name) {
+  jclass cls = (*env)->NewGlobalRef(env, bsg_safe_find_class(env, name));
+  if (cls == NULL) {
+    BUGSNAG_LOG("Could not load class %s", name);
+  }
+  return cls;
+}
+
+static inline jclass safe_get_method_id(JNIEnv *env, jclass cls, const char *name, const char *sig) {
+  jmethodID method = bsg_safe_get_method_id(env, cls, name, sig);
+  if (method == NULL) {
+    BUGSNAG_LOG("Could not load method %s with signature %s", name, sig);
+  }
+
+  return method;
+}
+
+static inline jmethodID safe_get_static_method_id(JNIEnv *env, jclass cls, const char *name, const char *sig) {
+  jmethodID method = bsg_safe_get_static_method_id(env, cls, name, sig);
+  if (method == NULL) {
+    BUGSNAG_LOG("Could not load static method %s with signature %s", name, sig);
```

Comment:
Could this logging just be moved to `bsg_safe_get_static_method_id` (and other functions) instead?

Reply:
I have another commit on the mainline that would clash with it. I think we should eventually put the logging in the safe_jni section, but not right now for this fix.